### ISSUE TITLE
Huge log files in the home folder

### ISSUE THREAD

**Issue by andoruB** [arvidn/libtorrent#2065]
libtorrent version (or branch):

1.0.10.0

platform/architecture:

Debian Sid amd64

compiler and compiler version:

gcc 6.3.0

Initially asked this on [qBittorrent's git repo](https://github.com/qbittorrent/qBittorrent/issues/6937), but I was told it is a libtorrent issue, so I copied the issue here too.

I get quite large log files whenever I run qBittorrent in my home folder, I've even seen some of them get as big as a few GiB.
The files in question: disk_access.log, disk_buffers.log, disk_io_thread.log and (the larger of all 4) buffer_stats.log
There's also a "session_stats" folder that I've tried to prevent from being created by linking it to /dev/null, but all that did is to inflate my ~/.xsession-errors...
Is there any way to disable these useless logs from taking up space and causing excessive disk writing?

**Comment by arvidn** [comment]
if you build libtorrent *without* defining the macros to enable these logs, they won't be created. See: https://github.com/arvidn/libtorrent/blob/libtorrent-1_0_10/docs/building.rst

**Comment by arvidn** [comment]
such as ``TORRENT_VERBOSE_LOGGING``, ``TORRENT_LOGGING``, ``TORRENT_DISK_STATS``, ``TORRENT_STATS`` etc.

**Comment by andoruB** [comment]
Where do I define these?

**Comment by arvidn** [comment]
It depends on how you build libtorrent. How are you building libtorrent?

**Comment by andoruB** [comment]
Well, I've got the 1.0.10 release required by qBittorrent from here: https://github.com/arvidn/libtorrent/releases/download/libtorrent-1_0_10/libtorrent-rasterbar-1.0.10.tar.gz
Then run ./configure with parameters and then make.
Hope that answers your question.
I'm guessing that it's not enough to pass --disable-debug when configuring?

**Comment by arvidn** [comment]
I think the: "with parameters" part is relevant. Perhaps you build it with parameters to enable all that logging. What are the parameters you run ./configure with?

**Comment by arvidn** [comment]
```
$ tar xzf libtorrent-rasterbar-1.0.10.tar.gz
$ cd libtorrent-rasterbar-1.0.10
$ ./configure
checking whether make supports nested variables... yes

Building libtorrent-rasterbar 1.0.10
...
$ make V=1
Making all in include/libtorrent
make[1]: Nothing to be done for `all'.
Making all in src
depbase=`echo web_connection_base.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
	/bin/sh ../libtool  --tag=CXX   --mode=compile g++ -DPACKAGE_NAME=\"libtorrent-rasterbar\" -DPACKAGE_TARNAME=\"libtorrent-rasterbar\" -DPACKAGE_VERSION=\"1.0.10\" -DPACKAGE_STRING=\"libtorrent-rasterbar\ 1.0.10\" -DPACKAGE_BUGREPORT=\"arvid@libtorrent.org\" -DPACKAGE_URL=\"http://www.libtorrent.org\" -DPACKAGE=\"libtorrent-rasterbar\" -DVERSION=\"1.0.10\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DHAVE_PTHREAD=1 -DHAVE_BOOST=/\*\*/ -DHAVE_BOOST_SYSTEM=1 -DHAVE_GETHOSTBYNAME=1 -DHAVE_CLOCK_GETTIME=1 -DNDEBUG=1 -DTORRENT_USE_OPENSSL=1 -DWITH_SHIPPED_GEOIP_H=1 -DBOOST_ASIO_HASH_MAP_BUCKETS=1021 -DBOOST_EXCEPTION_DISABLE=1 -DBOOST_ASIO_ENABLE_CANCELIO=1 -DBOOST_ASIO_DYN_LINK=1 -DTORRENT_BUILDING_SHARED=1 -I.  -I../include -I../ed25519/src -Os -I/opt/local/include -I/usr/local/include   -g -O2 -ftemplate-depth=120 -fvisibility=hidden -fvisibility-inlines-hidden -MT web_connection_base.lo -MD -MP -MF $depbase.Tpo -c -o web_connection_base.lo web_connection_base.cpp &&\
	mv -f $depbase.Tpo $depbase.Plo
libtool: compile:  g++ -DPACKAGE_NAME=\"libtorrent-rasterbar\" -DPACKAGE_TARNAME=\"libtorrent-rasterbar\" -DPACKAGE_VERSION=\"1.0.10\" "-DPACKAGE_STRING=\"libtorrent-rasterbar 1.0.10\"" -DPACKAGE_BUGREPORT=\"arvid@libtorrent.org\" -DPACKAGE_URL=\"http://www.libtorrent.org\" -DPACKAGE=\"libtorrent-rasterbar\" -DVERSION=\"1.0.10\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DHAVE_PTHREAD=1 "-DHAVE_BOOST=/**/" -DHAVE_BOOST_SYSTEM=1 -DHAVE_GETHOSTBYNAME=1 -DHAVE_CLOCK_GETTIME=1 -DNDEBUG=1 -DTORRENT_USE_OPENSSL=1 -DWITH_SHIPPED_GEOIP_H=1 -DBOOST_ASIO_HASH_MAP_BUCKETS=1021 -DBOOST_EXCEPTION_DISABLE=1 -DBOOST_ASIO_ENABLE_CANCELIO=1 -DBOOST_ASIO_DYN_LINK=1 -DTORRENT_BUILDING_SHARED=1 -I. -I../include -I../ed25519/src -Os -I/opt/local/include -I/usr/local/include -g -O2 -ftemplate-depth=120 -fvisibility=hidden -fvisibility-inlines-hidden -MT web_connection_base.lo -MD -MP -MF .deps/web_connection_base.Tpo -c web_connection_base.cpp  -fno-common -DPIC -o .libs/web_connection_base.o
```

The only ``TORRENT_*`` macros defined are: ``TORRENT_BUILDING_SHARED`` and ``TORRENT_USE_OPENSSL``.

**Comment by andoruB** [comment]
Last time when I built libtorrent, I remember I only passed the `--enable-encryption` parameter, as `./configure --help` mentioned all debugging features are disabled by default. This time I explicitly disabled all features I found relevant: `./configure --disable-invariant-checks --disable-statistics --disable-disk-stats --enable-geoip --enable-encryption` and so far I haven't got those logs anymore.
Will close the issue. Thanks for your help and patience.
